doc/rst/tools/mason/mason.rst

```diff
@@ -159,202 +194,364 @@ To try out different values at runtime, pass the values for ``number`` to ``maso
+the ``.pc`` file of a particular package, use ``mason system pc <package>`` where ``<package>``
+is replaced with the particular package you are looking for. Here is an example of a workflow
+for creating a Mason project with ``openssl`` which has already been installed.
+
+First, search to see that it is installed with ``mason system search openSSl`` which outputs:
+
+.. code-block:: sh
+
+   $ mason system search openssl
+   openssl               OpenSSL - Secure Sockets Layer and cryptography libraries and tools
+
+To find out more about the package, since it is in fact installed on my system, use the
+``mason system pc`` command as follows
+
+.. code-block:: sh
+
+   $ mason system pc openssl
+
+   ------- openSSL.pc -------
+
+   prefix=/usr
+   exec_prefix=${prefix}
+   libdir=${exec_prefix}/lib
+   includedir=${prefix}/include
+
+   Name: OpenSSL
+   Description: Secure Sockets Layer and cryptography libraries and tools
+   Version: 0.9.8zh
+   Requires:
+   Libs: -L${libdir} -lssl -lcrypto -lz
+   Cflags: -I${includedir}
+
+   -------------------
+
+
+Use the ``mason add --system`` command to add the dependency to the Mason.toml of the project.
+
+.. code-block:: sh
+
+   $ mason add --system openSSL@0.9.8zh
+   Adding system dependency openSSL version 0.9.8zh
+
+The ``Mason.toml`` now looks like:
+
+.. code-block:: text
+
+   [brick]
+   name = "myProject"
+   version = "0.1.0"
+   chplVersion = "1.18.0"
+
+   [system]
+   openSSL = "0.9.8zh"
+
+Now, upon calling ``mason build`` or ``mason run --build``, Mason will go get ``openssl`` and include it
+in the project so that it can be used as a dependency.
+
+
+Using Spack Dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~
+
 Mason users can interface with ``Spack``, a package manager geared towards high performance
-computing, through the ``mason external`` command. Though this integration, Mason users now have
+computing. Though this integration, Mason users now have
 access to a large ecosystem of `packages <https://spack.readthedocs.io/en/latest/package_list.html#package-list>`_.
 Non-destructive installs, custom version and configurations, and simple package installation
 and uninstallation are a few of the features Mason gains through this integration.
 
-The following example uses the ``HDF5`` library which is used for high performance data I/O.
-The ``HDF5.chpl`` wrapper used in the example can be found in ``CHPL_HOME/modules/packages``.
-Before Mason, Chapel users were required to download and install the libraries themselves
-and use Make to include and use them in their projects. These requirements have been removed
-and now only a few Mason commands are required to include external dependencies such as ``HDF5``.
+Mason users can access Spack through the ``mason external`` command. Spack provides Mason users with the ability
+to install and use any package in the Spack registry. This interface is analogous to the previous example except
```

Comment:
>  Spack registry.

Can this be a hyperlink to the spack registry like above?

https://spack.readthedocs.io/en/latest/package_list.html#package-list